docs: Adding a .goosehints guide (#279)
block/goose: docs/guidance/using-goosehints.md

# Using `.goosehints` in Goose

> `.goosehints` are text files used within the Goose environment to provide additional context about your project and improve the communication between the developer and Goose. Ensuring it understands your requirements better and can execute tasks more effectively.

Reviewer: ```suggestion
`.goosehints` are text files used within the Goose environment to provide additional context about your project and improve the communication between the developer and Goose. The use of `goosehints` ensures that Goose understands your requirements better and can execute tasks more effectively.
```

>[!TIP]
> **Developer toolkit required**
>

> > To make use of the hints file, you need to have the `developer` toolkit enabled.

Reviewer: link to guide on how to enable

> This guide will walk you through using your `.goosehints` file to streamline your workflow with custom instructions and context.

Reviewer: ```suggestion
This guide will walk you through creating and using your `.goosehints` file to streamline your workflow with custom instructions and context.
```

## Creating your `.goosehints` file

> You can place a `.goosehints` file in your current working directory or globally at `~/.config/goose/.goosehints`. This file can include any repeated instructions or contextual details relevant to your projects.x

Reviewer 2: Looks like there's an x at the end of this sentence that may be unnecessary.

Author: ooh great catch thanks. A typo.

A good time to consider adding a `.goosehints` file is when you find your self repeating prompts, or providing the same kind of instructions multiple times.

> The `.goosehints` file follows [jinja templating rules][jinja-guide] in case you want to leverage templating to insert file contents or variables. But you can also add instructions in natural language.

Reviewer: ```suggestion
The `.goosehints` file follows [jinja templating rules][jinja-guide], so you can leverage templating to insert file contents or variables. You can also add instructions in natural language.
```

### Setting Rules

> 

Reviewer: maybe one sentence giving context on what a rule is exactly?

> Some rules you can define for Goose to follow:

Author: I think I might need to change the lingo here - because these aren't exactly rules. They are prompts/hints/information that Goose includes in all requests - so there can't be an extensive list. I'll rephrase this and re-request a review. @angiejones

- **Decision-Making**: Specify if Goose should autonomously make changes or confirm actions with you first.
- **Validation Routines**: Provide test cases or validation methods that Goose should perform to ensure changes meet project specifications.
- **Feedback Loop**: Include steps that allow Goose to receive feedback and iteratively improve its suggestions.
- **Point to more detailed documentation**: Indicate important files like `README.md`, `CONTRIBUTING.md`, or others that Goose should consult for detailed explanations.

> Example:

Reviewer: ```suggestion
Example Rule:
```

which type of rule is this?
